src/include/random.h

```diff
@@ -0,0 +1,115 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2017 SUSE LINUX GmbH
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+*/
+
+#ifndef CEPH_RANDOM_H
+#define CEPH_RANDOM_H 1
+
+#include <random>
+
+// Basic random number facility, adapted from N3551:
+namespace ceph {
+namespace util {
+
+namespace detail {
+
+template <typename T>
+T& urng()
+{
+ static T rng;
+ return rng;
+}
+
+} // namespace detail
+
+template <typename EngineT = std::default_random_engine>
+void randomize_rng()
+{
+ static std::random_device rd;
+ detail::urng<EngineT>().seed(rd());
+}
+
+template <typename EngineT = std::default_random_engine>
+void randomize_rng(const int seed)
+{
+ detail::urng<EngineT>().seed(seed);
+}
+
+// Generate a random number between 0 and maxint:
+template <int max = std::numeric_limits<int>::max(), 
+          int min = 0,
+          typename DistributionT = std::uniform_int_distribution<int>,
+          typename EngineT = std::default_random_engine>
+int generate_random_number()
+{
+ static DistributionT d { min, max };
+
+ return d(detail::urng<EngineT>());
+}
+
+namespace detail {
+
+template <typename NumberT,
+          typename DistributionT,
+          typename EngineT>
+NumberT generate_random_number(const NumberT max, const NumberT min)
+{
+ static DistributionT d { min, max };
```

Comment:
the distribution itself is trivial to construct, it just calls the param_type ctor